### DISCUSSION TITLE
what is ram requirements

### DISCUSSION THREAD

**Discussion by furllmm** [minio/minio#19133] [Q&A] [answered]
i have problem minio uses more ram overtime up to %100 ram usage uses like 3g ram
 i wondered ram requirements 
is there any chance for limit ram usage if possible

i have 8G ram  
laptop b/m: lenovo ideapad slim 3

**Comment by ravindk89** [comment, marked as answer]
https://min.io/docs/minio/linux/operations/checklists/hardware.html#memory

We have some light guidance on recommended memory based on host storage. 

As of https://github.com/minio/minio/releases/tag/RELEASE.2024-01-28T22-35-53Z we pre-allocate a minimum of 2Gi per node in distributed and 1Gi per node in single-node topologies. Add `set CI_CD=true` envvar to reduce this allocation to 256MiB. Persisting this value across restarts will depend on your operating system.

You could always run MinIO in a container and limit memory that way. 

```shell
podman run -dt                                  \
  -p 9000:9000 -p 9001:9001                     \
  -v PATH:/mnt/data                             \
  -v /etc/default/minio:/etc/config.env         \
  -memory "2Gi"                                 \
  --name "minio_local"                          \
  minio server --console-address ":9001"
```

Set all environment variables in the `/etc/default/minio` file, or whatever path to an envar file you want for your local system.

**Reply by ravindk89** [reply]
@furllmm please add your comments in a thread, otherwise it is difficult to assist.

> ‘set’ is not a minio sub-command. See ‘minio --help’.

You have not provided the full command you are attempting to run here

> i use exe server

Windows does support Podman/Docker, which may be an option.

Windows may also have some OS level settings for restricting memory usage. 

Setting Environment Variables in Windows also will require you to look up how to do that for the OS, I do not know it off the top of my head.

**Reply by furllmm** [reply]
my full cmd output
![image](https://github.com/minio/minio/assets/73689928/33fd989d-984d-4ae4-abe1-c566ee6ed00c)

**Reply by harshavardhana** [reply]
you need to do 

```
set CI=true
minio.exe server A:\minio
```

**Reply by furllmm** [reply]
ty done

**Reply by xujuchao** [reply]
@harshavardhana Does the linux system set MEMORY to 4G directly after minio.conf

**Reply by xujuchao** [reply]
What does set CI=true mean?

**Reply by wuyudian1** [reply]
In the source code, we can find the answer:
./cmd/erasure-server-pool.go
```go
	if globalIsCICD || strconv.IntSize == 32 {
		n = 256 // 256MiB for CI/CD environments is sufficient or on 32bit platforms.
	}
```

**Reply by xujuchao** [reply]
Does that mean 64-bit platforms don't need to be set up?

**Reply by wuyudian1** [reply]
这个和系统是多少位的没关系，源码注释中用的是 or，表示2中条件中的任何一种满足，都会设置n = 256。
如果你不想预先分配那么大的内存，不用管到底是32位系统还是64位系统，你直接设置 CI = true，就能确保 n = 256

**Reply by xujuchao** [reply]
明白了谢谢 @wuyudian1
